### PR TITLE
chore(infra): bootstrap Specorator workflow shell

### PR DESCRIPTION
**Pull request by Luis85**
## Summary

- **ADR-001 violation fixed** — `useFeatures` was constructing `FeatureRepository` directly (5x per composable, once per operation). UI now depends on `FeatureService` (application layer) via Vue provide/inject; zero infrastructure imports remain in the composable.
- **`FeatureRepository` constructor hardened** — replaced `() => PluginSettings` getter with `SettingsPort`; each method reads live settings via `await settingsPort.getSettings()` instead of a closure over a stale fetch.
- **Two shallow modules deleted** — `UseCase<TInput,TOutput>` (marker interface, no behavior) and `errorMessages.ts` (single-entry lookup table); the one meaningful mapping inlined into `FeedbackService`.

## Files changed

| File | Change |
|---|---|
| `src/application/feature/FeatureService.ts` | New — wraps 5 use cases behind a clean string-param interface |
| `src/ui/composables/useFeatureService.ts` | New — FEATURE_SERVICE_KEY + useFeatureService() inject wrapper |
| `src/infrastructure/bridge/FeatureRepository.ts` | Constructor: () => PluginSettings replaced by SettingsPort |
| `src/ui/composables/useFeatures.ts` | 115 to 85 lines; no infrastructure imports |
| `src/plugin/SpecoratorView.ts` | Provides FeatureService at mount |
| `src/ui/main.ts` | Provides FeatureService at mount (standalone) |
| `src/plugin/main.ts` | MCP repo: () => this.settings replaced by this.bridge |
| `src/application/shared/UseCase.ts` | Deleted |
| `src/application/shared/errorMessages.ts` | Deleted |
| `tests/application/shared/errorMessages.test.ts` | Deleted |
| All MCP adapter tests + Home.test.ts + useFeatures.test.ts | Harness updates |

## Verification

```
npm run typecheck   # clean
npm run test        # 575/575 passing
```

---

## How to proceed — remaining architecture candidates

These were surfaced in the same /specorator:improve-codebase-architecture session. Each is independent and can be its own PR.

### Candidate 2 — Wire FeedbackService into useFeatures (medium, ~1 h)

**What:** `useFeatures` has 5 identical error paths: `store.setError(result.error.message)`. `FeedbackService` exists for exactly this — coordinated log + notification + error label — but is bypassed.

**How:** Inject `useLoggerPort()` + `useNotificationPort()` into `useFeatures`, construct `FeedbackService` once at composable init, replace inline `store.setError(...)` with `feedbackService.reportResult(result, { operation, errorLabel })`. The `KNOWN_MESSAGES` mapping in `FeedbackService` then fires for domain validation errors (e.g. "Title cannot be empty" -> "Please enter a feature title.").

**Files:** `src/ui/composables/useFeatures.ts`, possibly add `useLoggerPort` to composable.

**Tests:** Update `useFeatures.test.ts` harness to also provide `LOGGER_PORT` + `NOTIFICATION_PORT`.

---

### Candidate 5 — Extract MCP tool logic into testable functions (large, ~3 h)

**What:** 6 `register*.ts` files (~900 lines) have tool logic written inline inside `mcp.registerTool(...)` callbacks — untestable in isolation. `vault_search` has a silent catch {} with no test; canvas parsing edge cases uncovered.

**How:** Extract each tool handler body into a plain function (e.g. `searchVault(query, folder, vault): Promise<...>`). The `register*.ts` files become thin wiring. Unit tests hit the extracted functions directly; integration tests verify registration.

**Files:** `src/infrastructure/obsidian/mcp/register*.ts` (6 files), new `tests/infrastructure/mcp/` test files.

**Start point:** `registerVaultAndFeatureTools.ts` — highest complexity, has the uncovered catch {}.

---

### Candidate 6 — ProposalStore unit tests (small, ~1 h)

**What:** `ProposalStore` is the ADR-007 review boundary — no agent write reaches the vault without user acceptance. Its state machine (pending -> accepted / rejected) has zero unit tests; correctness is only inferred from MCP integration tests.

**How:** Add `tests/infrastructure/proposal-store.test.ts` covering: queue() creates pending proposal, accept() invokes mutate + transitions to accepted, reject() skips mutate + transitions to rejected, double-accept guard, exception during mutate() reverts to pending.

**Note:** `tests/infrastructure/proposal-store.test.ts` may already exist — check coverage before adding.

---

### Priority order

1. **Candidate 2** (FeedbackService wiring) — small, completes the error-reporting triad, natural follow-on
2. **Candidate 6** (ProposalStore tests) — small, high safety value for the agent boundary
3. **Candidate 5** (MCP tool extraction) — largest, best done as its own focused PR

Generated with Claude Code

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -6,6 +6,53 @@ If a rule here disagrees with `docs/contributing.md`, the human guide wins; open
 
 ---
 
+## 0. Read these first
+
+1. **`memory/constitution.md`** — governing principles. Override only with explicit human approval.
+2. **`.claude/memory/MEMORY.md`** — operational memory: workflow rules + project state, indexed.
```
[the remark this answers is not in the document]
Comment (Luis85) [reply AGENTS.md]: Fixed: `*(optional)*` qualifier added to item 2. The load condition (Load

File: .codex/workflows/pr-delivery.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+4. Install the project inside the new worktree before running verification:
+
+   ```bash
+   npm ci
+   ```
+
+   Fresh worktrees do not share `node_modules` with the main checkout. Skipping this step can make first-run verify actions fail before they reach the actual project checks.
+5. Work only inside that worktree.
+6. Keep the change to one concern and update the corresponding docs, state files, or issue records in the same branch.
+7. Run the relevant verification gate before pushing. Use `npm run verify` as the final gate. During iteration, `specorator check:fast` (typecheck + tests + registry + agents), `specorator check:content` (links, ADR index, docs), and `specorator check:workflow` (specs, roadmaps, traceability) are available for narrower feedback. Run targeted checks and state exactly what passed.
```
[the remark this answers is not in the document]
Comment (Luis85) [reply .codex/workflows/pr-delivery.md]: Already fixed in a subsequent commit. The current file uses `npm run verify`, `npm run verify:scaffold`, and `npm run verify:workflows` — all defined in `package.json`. No `specorator check:*` commands appear in the current version.

File: AGENTS.md [hunk 3]
```diff
@@ -6,6 +6,53 @@ If a rule here disagrees with `docs/contributing.md`, the human guide wins; open
 
 ---
 
+## 0. Read these first
+
+1. **`memory/constitution.md`** — governing principles. Override only with explicit human approval.
+2. **`.claude/memory/MEMORY.md`** *(agent-local, not in repo)* — operational memory index maintained by Claude Code per machine. Load if present; skip if absent.
+3. **`docs/specorator.md`** — full Specorator workflow definition (Stages 1–11 + opt-in tracks).
```
[the remark this answers is not in the document]
Comment (Luis85) [reply AGENTS.md]: Fixed: item 3 now reads `*(optional — scaffold stub, not yet populated; skip until filled in)*` and its description reframes it as a "future workflow definition stub" rather than an active canonical source.

File: CLAUDE.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # CLAUDE.md
 
+@AGENTS.md
+@memory/constitution.md
+@.claude/memory/MEMORY.md
```
[the remark this answers is not in the document]
Comment (Luis85) [reply CLAUDE.md]: Already fixed in a subsequent commit. The current `CLAUDE.md` imports only `@AGENTS.md` and `@memory/constitution.md` — both files tracked in-repo. There is no `@.claude/memory/MEMORY.md` hard-import. The `.claude/memory/MEMORY.md` reference in `AGENTS.md` item 2 is clearly marked optional ("Load if present; skip if absent").

File: templates/project-review-plan-template.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+## Evidence plan
+
+| Source | Query or path | Why it matters | Status |
+|---|---|---|---|
+| Workflow state | `specs/<slug>/workflow-state.md` | Stage progress and handoffs | pending |
+| Git history | `<range>` | Commit cadence and branch hygiene | pending |
+| Pull requests | `<query>` | Review feedback and rework | pending |
+| CI | `<query>` | Verification failures and flaky checks | pending |
+| Issue mirror sync | `npm run sync:issues -- --dry-run --json` or `skipped: <reason>` | Advisory comparison between local `issues/` records and GitHub issue metadata | pending |
```
[the remark this answers is not in the document]
Comment (Luis85) [reply templates/project-review-plan-template.md]: Already addressed in the current template. The Issue mirror evidence rows use "project-specific command or `skipped: <reason>`" — running `npm run sync:issues` / `npm run check:issues` is explicitly skippable with a documented reason. No `missing script` error would occur if the reviewer follows the template as written.

File: .codex/instructions.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+1. `AGENTS.md`
+2. `memory/constitution.md`
+3. `.claude/memory/MEMORY.md` *(agent-local, not tracked in repo — load if present, skip if absent)*
+4. The relevant workflow doc under `docs/`
+5. The relevant Codex workflow under `.codex/workflows/`
+
+Load scoped steering docs from `docs/steering/` only when they matter to the task.
+
+For architecture-improvement work — reducing coupling, refactoring shallow modules, or improving testability / AI-navigability — also read `.claude/skills/improve-codebase-architecture/SKILL.md` before proposing edits. Treat it as a practice guide: surface deepening opportunities first, then make implementation changes only after a candidate is selected.
```
[the remark this answers is not in the document]
Comment (Luis85) [reply .codex/instructions.md]: The current instruction already says "read the `improve-codebase-architecture` skill **if available** (shipped by the specorator plugin; not tracked in this repo)" — this is conditional on skill availability. Agents that don't have the skill file skip the step.

File: templates/project-review-history-template.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+| Item | Signal | Evidence | Notes |
+|---|---|---|---|
+| #<n> |  |  |  |
+
+## Issue mirror review
+
+| Check | Command or source | Result | Notes |
+|---|---|---|---|
+| Sync dry-run | `npm run sync:issues -- --dry-run --json` | pending | Advisory; skip with reason when `gh` access or local mirror records are unavailable. |
+| Drift check | `npm run check:issues` | pending | Advisory; warnings such as missing linked issue files are review evidence, not universal verify failures. |
```
[the remark this answers is not in the document]
Comment (Luis85) [reply templates/project-review-history-template.md]: Already addressed. Both rows in the current template are labelled "project-specific `npm run <cmd>` **if defined**" and classified as Advisory. A missing script means skip-with-reason, not a blocking error.

File: .codex/workflows/pr-delivery.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
+# PR delivery workflow
+
+This repo treats Codex as an autonomous topic-branch contributor. When a human asks Codex to make a non-trivial repo change, Codex should complete the full PR loop unless the human explicitly asks for local-only work.
+
+## Default delivery loop
+
+1. Read the required project context from `AGENTS.md` and `.codex/instructions.md`.
+2. Check the main checkout is clean enough to start and identify the integration branch (`main` for Shape A, `develop` for Shape B).
```
[the remark this answers is not in the document]
Comment (Luis85) [reply .codex/workflows/pr-delivery.md]: Fixed: `origin/<integration-branch>` in the worktree-add command is now pinned to `origin/develop`. The surrounding prose already named `develop` as the integration branch; the command now matches.

File: templates/onboarding/04-open-a-pr-and-enter-the-feedback-loop.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+## Steps
+
+1. **Create a topic branch** from your integration branch. Detect it (defaults to `main` if `origin/HEAD` is not set):
+   ```bash
+   INTEGRATION=$(git symbolic-ref --short refs/remotes/origin/HEAD 2>/dev/null | sed 's|^origin/||')
+   INTEGRATION=${INTEGRATION:-develop}
+   git switch "$INTEGRATION"
+   git pull --ff-only
+   git switch -c feat/<scope>/<short-description>
```
[the remark this answers is not in the document]
Comment (Luis85) [reply templates/onboarding/04-open-a-pr-and-enter-the-feedback-loop.md]: Already correct in the current version. Branch creation uses `feature/<short-kebab>` with an explicit note listing the allowed prefixes (`feature`, `fix`, `docs`, `chore`, `refactor`), matching `AGENTS.md` §4. The commit/PR title uses Conventional Commits format (`feat(<scope>): ...`) which is a separate thing from the branch name.

File: templates/onboarding/01-create-your-first-issue.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+## Acceptance criteria
+
+- [ ] Issue title follows the `type(scope): description` format.
+- [ ] Issue body contains a Problem section, a Proposed solution section, and a checklist of acceptance criteria.
+- [ ] At least one type label and one priority label are applied.
+- [ ] `status:draft` label is applied.
+
+## Next step
+
+Once your issue is created, move to [Issue 2 — Triage: classify and prioritise]().
```
[the remark this answers is not in the document]
Comment (Luis85) [reply templates/onboarding/01-create-your-first-issue.md]: Already fixed in a subsequent commit. All "Next step" links in the onboarding series resolve to real file names (e.g. `02-triage-classify-and-prioritise.md`). No empty `[]()` targets remain in the current version.

File: manifest.json [hunk 11]
```diff
@@ -2,7 +2,7 @@
 	"id": "specorator",
 	"name": "Specorator",
 	"version": "0.0.1",
-	"minAppVersion": "1.7.2",
+	"minAppVersion": "1.12.7",
```
[the remark this answers is not in the document]
Comment (Luis85) [reply manifest.json:5]: `minAppVersion 1.12.7` is an intentional project policy decision, not a side-effect of this bootstrap commit. The version floor was set deliberately for this project's compatibility target independent of specific API requirements introduced here.

[the remark this answers is not in the document]
Comment (Luis85) [reply manifest.json:5]: Same as the other thread on this line: `minAppVersion 1.12.7` is intentional project policy. The compatibility floor was set deliberately, not as a side-effect of this bootstrap commit.
